File: JavaScript/insertionSort.js
function insertionSort(inputArr) {
    console.log("Unsorted array:", inputArr);
    let A = inputArr.length;
    
    for (let j=1; j < A; j++) {
        // current element in unsorted subarray
        let currentValue = inputArr[j];
        // last element of sorted subarray
        let i = j - 1;

        while ((i > -1) && (currentValue < inputArr[i])) {
            inputArr[i+1] = inputArr[i];
            i = i - 1;
        };
        inputArr[i+1] = currentValue;
    };
    return console.log("Sorted array:", inputArr);
};

export default insertionSort;

File: Rust/insertion_sort.rs
pub fn insertion_sort<T: Ord + Copy> (arr: &mut[T]) {

    for i in 1..arr.len() {
        let mut j = i;
        let current = arr[i];

        while j > 0 && current < arr[j - 1] {
            arr[j] = arr[j - 1];
            j -= 1;
        }

        arr[j] = current;
    }
}


fn main() {

    let mut arr = vec![5,2,4,6,1,3];
    
    println!("Unsorted array: {:?}", arr);

    insertion_sort(&mut arr);
    println!("Sorted array: {:?}", &arr);

}
